src/docfx/build/page/BuildPage.cs

```diff
@@ -133,21 +135,46 @@ private static (List<Error> errors, string pageType, object content, JObject met
                 schema = Path.GetFileNameWithoutExtension(schema);
             }
 
-            return LoadSchemaDocument(errors, token, schema);
+            return LoadSchemaDocument(errors, token, schema, file);
         }
 
         private static (List<Error> errors, string pageType, object content, JObject metadata)
             LoadSchemaDocument(
-            List<Error> errors, JToken token, string schema)
+            List<Error> errors, JToken token, string schema, Document file)
         {
             if (schema == null || !s_schemas.TryGetValue(schema, out var schemaType))
             {
                 throw Errors.SchemaNotFound(schema).ToException();
             }
 
-            var content = token.ToObject(schemaType);
-
+            var (schemaViolationErrors, content) = JsonUtility.ToObject(token, schemaType, TransformContent);
+            errors.AddRange(schemaViolationErrors);
             return (errors, schema, content, token.Value<JObject>("metadata"));
+
+            object TransformContent(SchemaContentTypeAttribute attribute, JsonReader reader)
+            {
+                // Schema violation if the field is not what SchemaContentTypeAttribute required
+                if (reader.ValueType != attribute.RequiredType)
+                {
+                    var lineInfo = reader as IJsonLineInfo;
+                    var range = new Range(lineInfo.LineNumber, lineInfo.LinePosition);
+                    errors.Add(Errors.ViolateSchema(range, $"Field with attribute '{attribute.GetType().ToString()}' should be of type {attribute.RequiredType.ToString()}."));
```

Comment:
range is not passed to error

Reply:
you mean line property of error? set inside the ViolateSchema error

---
In reply to: [210854133](https://github.com/dotnet/docfx/pull/3209#discussion_r210854133) [](ancestors = 210854133)